samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,694 @@
+ *
+ * Element 0 Root Models
+ */
+
+static struct bt_mesh_model root_models[] = {
+	BT_MESH_MODEL_CFG_SRV(&cfg_srv),
+	BT_MESH_MODEL_CFG_CLI(&cfg_cli),
+	BT_MESH_MODEL_HEALTH_SRV(&health_srv, &health_pub),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op,
+		      &gen_onoff_pub_srv, &onoff_state[0]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli, &onoff_state[0]),
+};
+
+/*
+ * Element 1 Models
+ */
+
+static struct bt_mesh_model secondary_0_models[] = {
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op,
+		      &gen_onoff_pub_srv_s_0, &onoff_state[1]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli_s_0, &onoff_state[1]),
+};
+
+/*
+ * Element 2 Models
+ */
+
+static struct bt_mesh_model secondary_1_models[] = {
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op,
+		      &gen_onoff_pub_srv_s_1, &onoff_state[2]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli_s_1, &onoff_state[2]),
+};
+
+/*
+ * Element 3 Models
+ */
+
+static struct bt_mesh_model secondary_2_models[] = {
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_SRV, gen_onoff_srv_op,
+		      &gen_onoff_pub_srv_s_2, &onoff_state[3]),
+	BT_MESH_MODEL(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, gen_onoff_cli_op,
+		      &gen_onoff_pub_cli_s_2, &onoff_state[3]),
+};
+
+/*
+ * Button to Client Model Assignments
+ */
+
+struct bt_mesh_model *mod_cli_sw[] = {
+		&root_models[4],
+		&secondary_0_models[1],
+		&secondary_1_models[1],
+		&secondary_2_models[1],
+};
+
+/*
+ * LED to Server Model Assigmnents
+ */
+
+struct bt_mesh_model *mod_srv_sw[] = {
+		&root_models[3],
+		&secondary_0_models[0],
+		&secondary_1_models[0],
+		&secondary_2_models[0],
+};
+
+/*
+ * Root and Secondary Element Declarations
+ */
+
+static struct bt_mesh_elem elements[] = {
+	BT_MESH_ELEM(0, root_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_0_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_1_models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(0, secondary_2_models, BT_MESH_MODEL_NONE),
+};
+
+static const struct bt_mesh_comp comp = {
+	.cid = CID_INTEL,
+	.elem = elements,
+	.elem_count = ARRAY_SIZE(elements),
+};
+
+struct device *sw_device;
+
+struct sw {
+	u32_t pin;
+	struct k_work button_work;
+};
+
+static struct sw sw;
+
+static struct gpio_callback button_cb;
+
+static u8_t trans_id;
+static u32_t time, last_time;
+static u16_t primary_addr;
+static u16_t primary_net_idx;
+
+/*
+ * Generic OnOff Model Server Message Handlers
+ *
+ * Mesh Model Specification 3.1.1
+ *
+ */
+
+static void gen_onoff_get(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(2 + 1 + 4);
+	struct onoff_state *onoff_state = model->user_data;
+
+	printk("%s addr: 0x%04x onoff: 0x%02x\n"
+			, __func__, model->elem->addr, onoff_state->current);
```

Comment:
The alignment of the split line is wrong, also the line split should always be after the ',' and never before it.

Reply:
ok